### PR TITLE
[FIX] analytic: plan's applicability don't work without companies

### PR DESCRIPTION
**Pull request by gawa-odoo**
If you create an applicability and remove the company field, they are never used.
An applicability like this should be valid for all companies.
We put a 0.5 value for the company field so an applicability so it has a lesser priority than other fields.
Same idea as the distribution models.

opw-3847415

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/tests/test_account_analytic.py [hunk 1]
```diff
@@ -264,3 +264,28 @@ def test_cross_analytics_computing(self):
         self.assertEqual(self.analytic_account_a.invoice_count, 1)
         self.analytic_account_a._compute_vendor_bill_count()
         self.assertEqual(self.analytic_account_a.vendor_bill_count, 1)
+
+    def test_applicability_score(self):
```
Comment (RicGR98) [inline addons/account/tests/test_account_analytic.py:268]: Perhaps a small preliminary explanation in the docstring could be helpful ?

**Review by RicGR98 — COMMENTED** [review]
Thanks for the PR :)

File: addons/analytic/tests/test_plan_operations.py [hunk 2]
```diff
@@ -76,8 +76,7 @@ def test_validate_company_plans(self):
             'name': 'Mandatory Plan',
             'default_applicability': 'optional',
         }])
-        mandatory_plan.with_company(company_2).write({'default_applicability': 'mandatory'})
```
[the remark this answers is not in the document]
Comment (gawa-odoo) [reply addons/analytic/tests/test_plan_operations.py:79]: Well it wasn't quite mandatory, it was only mandatory for self.env.company
And either this definition, or the applicability defined below was completely useless as they both did the same thing (on the only thing that was tested).
To be able to test other situations, I removed the default_applicability.
Doing so, I now changed the name of the variable, to reflect the difference

**Review by RicGR98 — APPROVED** [review]

File: addons/account/tests/test_account_analytic.py [hunk 3]
```diff
@@ -264,3 +264,29 @@ def test_cross_analytics_computing(self):
...
+            },
+        ])
+
+        applicability = self.default_plan._get_applicability(business_domain='invoice', company_id=self.env.company.id, product=self.product_a.id)
+        self.assertEqual(applicability, 'mandatory', "product takes precedence over company")
+
+        score = applicability_without_company._get_score(business_domain='invoice', product=self.product_a.id)
+        self.assertEqual(score, 2, "company_id shouldn't be taken into account")
+        score = applicability_with_company._get_score(business_domain='invoice', product=self.product_a.id)
+        self.assertEqual(score, 1, "company_id shouldn't be taken into account either")
```
Comment (oco-odoo) [inline addons/account/tests/test_account_analytic.py:292]: I find the message a bit misleading here (especially the "either"). Not sure those two (this one and the one above) make the test any more explicit. Wouldn't it be just better, to leave those two assertEqual without error message ? The "product takes precedence over company" above already states the expected behavior, if I get the test well :thinking:

**Review by oco-odoo — CHANGES_REQUESTED** [review]

File: addons/account/tests/test_account_analytic.py (hunk 3 above)
Comment (gawa-odoo) [reply addons/account/tests/test_account_analytic.py]: Not exactly, we're not really testing the same thing.
But still, I reworded the message to make clearer what I wanted to test in these 2 asserts as indeed, it wasn't explicit

File: addons/analytic/tests/test_plan_operations.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (oco-odoo) [reply addons/analytic/tests/test_plan_operations.py:79]: Okaaaay, got it ! Thanks :)

**Review by oco-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**